File: app/src/main/java/com/eddief/android/wallofaww/model/RedditAPI.java
package com.eddief.android.wallofaww.model;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface RedditAPI {

    @GET("/r/{subs}/.json?limit=100")
    Call<RedditPOJO> retrieveData(
            @Path("subs") String subs,
            @Query("after") String id
    );
}

File: app/src/main/java/com/eddief/android/infiniteaww/data/RedditAPI.kt
package com.eddief.android.infiniteaww.data

import com.eddief.android.infiniteaww.model.RedditPOJO
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface RedditAPI {

    companion object {
        private const val LIMIT = 100
    }

    @GET("/r/{subs}/.json?limit=$LIMIT")
    fun retrieveData(
        @Path("subs") subs: String,
        @Query("after") id: String?
    ): Call<RedditPOJO>
}